File: android/src/main/java/com/tradplus/flutter/TPResult.java
package com.tradplus.flutter;

import android.text.TextUtils;

import com.tradplus.ads.base.bean.MixAdInfo;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TPResult {
    private final com.tradplus.ads.mgr.TPResult mNativeResult;

    public TPResult() {
        mNativeResult = new com.tradplus.ads.mgr.TPResult();
    }

    public List<String> handleAdUnitId(List<String> adUnitIds) {
        return mNativeResult.handleAdUnitId(adUnitIds);
    }

    public List<Map<String, Object>> handleMix(List<Map<String, Object>> mixAdInfoList) {
        List<MixAdInfo> mixList = toMixAdInfoList(mixAdInfoList);
        List<MixAdInfo> result = mNativeResult.handleMix(mixList);
        return toMapList(result);
    }

    private List<MixAdInfo> toMixAdInfoList(List<Map<String, Object>> mixAdInfoList) {
        ArrayList<MixAdInfo> mixList = new ArrayList<>();
        if (mixAdInfoList == null) {
            return mixList;
        }
        for (Map<String, Object> map : mixAdInfoList) {
            if (map == null) {
                continue;
            }
            MixAdInfo info = new MixAdInfo();
            Object adUnitId = map.get("adUnitId");
            if (adUnitId instanceof String && !TextUtils.isEmpty((String) adUnitId)) {
                info.setAdUnitId((String) adUnitId);
            } else {
                Object ecpm = map.get("ecpm");
                if (ecpm instanceof Number) {
                    info.setEcpm(((Number) ecpm).doubleValue());
                }
            }
            mixList.add(info);
        }
        return mixList;
    }

    private List<Map<String, Object>> toMapList(List<MixAdInfo> mixAdInfoList) {
        ArrayList<Map<String, Object>> mapList = new ArrayList<>();
        if (mixAdInfoList == null) {
            return mapList;
        }
        for (MixAdInfo info : mixAdInfoList) {
            if (info == null) {
                continue;
            }
            HashMap<String, Object> map = new HashMap<>();
            String adUnitId = info.getAdUnitId();
            if (!TextUtils.isEmpty(adUnitId)) {
                map.put("adUnitId", adUnitId);
            } else {
                map.put("ecpm", info.getEcpm());
            }
            mapList.add(map);
        }
        return mapList;
    }
}

File: lib/tp_result.dart
import 'package:tradplus_sdk/tradplus_sdk.dart';

final tpResult = TPResult();

class TPResult {
  /// 多条 TradPlus 广告位 ID 按缓存 eCPM 从高到低排序
  Future<List<String>> handleAdUnitId(List<String> adUnitIds) async {
    final result = await TradplusSdk.channel.invokeMethod(
      'tpresult_handleAdUnitId',
      {'adUnitIds': adUnitIds},
    );
    if (result == null) {
      return [];
    }
    return List<String>.from(result);
  }

  /// TradPlus 广告位与第三方自定义价格混合比价，结果从高到低排序
  /// mixAdInfoList 每项传入 adUnitId 或 ecpm 其一即可
  Future<List<Map<String, dynamic>>> handleMix(
      List<Map<String, dynamic>> mixAdInfoList) async {
    final result = await TradplusSdk.channel.invokeMethod(
      'tpresult_handleMix',
      {'mixAdInfoList': mixAdInfoList},
    );
    if (result == null) {
      return [];
    }
    return (result as List)
        .map((item) => Map<String, dynamic>.from(item as Map))
        .toList();
  }
}
